apps/ewallet/lib/ewallet/gates/blockchain_transaction_gate.ex

```diff
@@ -117,6 +117,42 @@ defmodule EWallet.BlockchainTransactionGate do
     end
   end
 
+  def mint_erc20_token(
+        %{
+          from: from,
+          contract_address: contract_address,
+          amount: amount
+        },
+        rootchain_id
+      ) do
+    with :ok <- validate_rootchain_identifier(rootchain_id),
```

Comment:
Since there is only one condition does it make sense to turn this into a case statement?

Reply:
Maybe add a spec to this function too?